src/formats/SMI.cpp

```diff
@@ -0,0 +1,705 @@
+    double charge;
+    int charge_int = 0;
+    if (std::modf(atom.charge(), &charge) == 0.0 && charge != 0.0) {
+        needs_brackets = true;
+        charge_int = static_cast<int>(charge);
+    }
+
+    // Before the atom is printed, we must know if we are printing a property set.
+    if (explicit_h != 0.0 || smi_class || !chirality.empty() ||
+        aliphatic_organic.count(type) == 0) {
+        needs_brackets = true;
+    }
+
+    auto is_aromatic = atom.get<Property::BOOL>("is_aromatic").value_or(false);
+
+    if (is_aromatic) {
+        // We need to print brackets for aromatic Te, Se, As, and Si
+        // Note, we allow aromatic Bromine to be bare (no brackets) in following
+        // ChemAxon's standard.
+        if (type.size() > 1) {
+            needs_brackets = true;
+        }
+        std::transform(type.begin(), type.end(), type.begin(), ::tolower);
+    }
+
+    if (needs_brackets) {
+        fmt::print(*file_, "[");
+    }
+
+    // Mass must be first, before the element is printed
+    if (mass_int != 0) {
+        fmt::print(*file_, "{}", mass_int);
+    }
+
+    fmt::print(*file_, "{}", type);
+
+    if (smi_class && smi_class->kind() == Property::DOUBLE) {
+        fmt::print(*file_, ":{}", static_cast<int>(smi_class->as_double()));
+    }
+
+    if (chirality != "") {
+        // If it's clockwise, then we just print the symbol @@ and be done with it
+        if (chirality.find("CW") == 0) {
+            fmt::print(*file_, "@@");
+        } else {
+            fmt::print(*file_, "@");
+            // TODO check if the chirality flag is valid instead of trusting the user
+            if (chirality.size() > 4) {
+                fmt::print(*file_, "{}", chirality.substr(4));
+            }
+        }
+    }
+
+    if (explicit_h == 1.0) {
+        fmt::print(*file_, "H");
+    }
+
+    if (explicit_h > 1.0) {
+        fmt::print(*file_, "H{}", static_cast<int>(explicit_h));
+    }
+
+    if (charge_int < 0) {
+        fmt::print(*file_, "-", charge_int);
+    }
+
+    if (charge_int > 0) {
+        fmt::print(*file_, "+", charge_int);
+    }
+
+    charge_int = std::abs(charge_int);
+    if (charge_int != 1 && charge_int != 0) {
+        fmt::print(*file_, "{}", charge_int);
+    }
+
+    if (needs_brackets) {
+        fmt::print(*file_, "]");
+    }
+}
+
+static void print_bond(std::unique_ptr<TextFile>& file_, chemfiles::Bond::BondOrder bo) {
+    switch(bo) {
+    case Bond::SINGLE: break;
+    case Bond::DOUBLE: fmt::print(*file_, "="); break;
+    case Bond::TRIPLE: fmt::print(*file_, "#"); break;
+    case Bond::QUADRUPLE: fmt::print(*file_, "$"); break;
+    case Bond::AROMATIC: fmt::print(*file_, ":"); break;
+    case Bond::DATIVEL: fmt::print(*file_, "<-"); break;
+    case Bond::DATIVER: fmt::print(*file_, "->"); break;
+    case Bond::UP: fmt::print(*file_, "/"); break;
+    case Bond::DOWN: fmt::print(*file_, "\\"); break;
+    case Bond::UNKNOWN:
+    default:
+        fmt::print(*file_, "~");
```

Comment:
How about using `~` for UNKNOWN bonds, and adding a warning in the default case, to future proof this a bit?